src/google-maps/map-directions-renderer/map-directions-service.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+// Workaround for: https://github.com/bazelbuild/rules_nodejs/issues/1265
+/// <reference types="googlemaps" />
+
+import {Injectable, NgZone} from '@angular/core';
+import {Observable} from 'rxjs';
+
+export interface MapDirectionsResponse {
+  status: google.maps.DirectionsStatus;
+  result?: google.maps.DirectionsResult;
+}
+
+/**
+ * Angular service that wraps the Google Maps DirectionsService from the Google Maps JavaScript
+ * API.
+ *
+ * See developers.google.com/maps/documentation/javascript/reference/directions#DirectionsService
+ */
+@Injectable()
+export class MapDirectionsService {
+  private readonly _directionsService: google.maps.DirectionsService;
+
+  constructor(private readonly _ngZone: NgZone) {
+    this._directionsService = new google.maps.DirectionsService();
```

Comment:
Maybe this should be called once `route` is invoked for the first time?

Reply:
I can do this. What's the benefit though?

Reply:
Whatever cost there is to creating the service will be incurred only once the user has requested it. That being said, I don't know if there actually is any cost to it.

Reply:
Looking at the internals, it looks like merely instantiating the DirectionsService incurs no cost.